### PR TITLE
proc does not take untyped/typed as argument 

### PR DESCRIPTION
**Pull request by nc-x**
Fixes https://github.com/nim-lang/Nim/issues/6970
Fixes https://github.com/nim-lang/Nim/issues/7469
Closes https://github.com/nim-lang/Nim/issues/9669
Fixes https://github.com/nim-lang/Nim/issues/9593

### CODE REVIEW

**Comment by nc-x** [comment]
Ready for review.

**Comment by Araq** [comment]
Much simpler logic:

```nim
if owner.kind != skModule or owner.owner.name.s != "stdlib"
```

**Comment by nc-x** [comment]
Done.

File: compiler/semtypes.nim [hunk 1]
```diff
@@ -1054,6 +1054,10 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
 
     if hasType:
       typ = semParamType(c, a.sons[length-2], constraint)
+      var owner = getCurrOwner(c).owner
+      if (owner.kind != skModule or owner.owner.name.s != "stdlib") and
```
Comment (dom96) [inline compiler/semtypes.nim]: Why do you disable this for the stdlib?

Comment (nc-x) [reply compiler/semtypes.nim]: Done.

**Comment by krux02** [comment]
I just checked if the current development branch detects all examples that I had, and this one is still uncaught with your PR:

```nim
proc bar(): untyped =
  discard
```

Will you come up with another fix for that?

**Pull request merged**